include/zephyr/drivers/adc.h

```diff
@@ -146,6 +146,200 @@ struct adc_channel_cfg {
 #endif /* CONFIG_ADC_CONFIGURABLE_INPUTS */
 };
 
+/**
+ * @brief Get ADC channel configuration from a given devicetree node.
+ *
+ * This returns a static initializer for a <tt>struct adc_channel_cfg</tt>
+ * filled with data from a given devicetree node.
+ *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ * &adc0 {
+ *    #address-cells = <1>;
+ *    #size-cells = <0>;
+ *
+ *    channel@0 {
+ *        reg = <0>;
+ *        zephyr,gain = "ADC_GAIN_1_6";
+ *        zephyr,reference = "ADC_REF_INTERNAL";
+ *        zephyr,acquisition-time = <ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 20)>;
+ *        bipolar;
+ *        zephyr,input-positive = <NRF_SAADC_AIN6>;
+ *        zephyr,input-negative = <NRF_SAADC_AIN7>;
+ *    };
+ * };
+ * @endcode
+ *
+ * Example usage:
+ *
+ * @code{.c}
+ * static const struct adc_channel_cfg ch0_cfg_dt =
+ *     ADC_CHANNEL_CFG_DT(DT_CHILD(DT_NODELABEL(adc0), channel_0));
+ *
+ * // Initializes 'ch0_cfg_dt' to:
+ * // {
+ * //     .channel_id = 0,
+ * //     .gain = ADC_GAIN_1_6,
+ * //     .reference = ADC_REF_INTERNAL,
+ * //     .acquisition_time = ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 20),
+ * //     .differential = true,
+ * //     .input_positive = NRF_SAADC_AIN6,
+ * //     .input-negative = NRF_SAADC_AIN7,
+ * // }
+ * @endcode
+ *
+ * @param node_id Devicetree node identifier.
+ *
+ * @return Static initializer for an adc_channel_cfg structure.
+ */
+#define ADC_CHANNEL_CFG_DT(node_id) { \
+	.channel_id       = DT_REG_ADDR(node_id), \
+	.gain             = DT_STRING_TOKEN(node_id, zephyr_gain), \
+	.reference        = DT_STRING_TOKEN(node_id, zephyr_reference), \
+	.acquisition_time = DT_PROP(node_id, zephyr_acquisition_time), \
+	.differential     = DT_PROP(node_id, bipolar), \
+IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS, \
+	(.input_positive  = DT_PROP(node_id, zephyr_input_positive), \
+	 .input_negative  = DT_PROP_OR(node_id, zephyr_input_negative, 0),)) \
+}
+
+/**
+ * @brief Container for ADC channel information specified in devicetree.
+ *
+ * @see ADC_IO_CHANNEL_DT_SPEC_GET_BY_IDX
+ * @see ADC_IO_CHANNEL_DT_SPEC_GET
+ */
+struct adc_io_channel_dt_spec {
```

Comment:
suggestion: why not just `adc_dt_spec`?

Reply:
Right. There's no need to unnecessarily lengthen the names here. I'll give it a try.

Reply:
Applied.